framework/grid/DataColumn.php

```diff
@@ -144,11 +144,21 @@ protected function getHeaderCellLabel()
         if ($this->label === null) {
             if ($provider instanceof ActiveDataProvider && $provider->query instanceof ActiveQueryInterface) {
                 /* @var $model Model */
-                $model = new $provider->query->modelClass;
+                if (is_subclass_of($provider->query->modelClass, 'yii\db\ActiveRecordInterface')) {
```

Comment:
the property `ActiveQuery::$modelClass` gets defined here http://www.yiiframework.com/doc-2.0/yii-db-activequerytrait.html#$modelClass-detail and the docs mention is an active record class.

@SilverFire so can we assume `$modelClass` is a subclass of `yii\db\ActiveRecordInterface`?